mapo/envs/navigation.py

```diff
@@ -92,13 +92,25 @@ def _transition_fn(self, state, action):
         if self._deceleration_zones:
             deceleration = self._deceleration()
 
-        noise = self._sample_noise()
-        next_state = state + (deceleration * action) + noise
-        return next_state
+        position = state + (deceleration * action)
+        dist, next_state = self._sample_noise(position)
+        log_prob = dist.log_prob(tf.stop_gradient(next_state))
+        return next_state, log_prob
+
+    def _transition_log_prob_fn(self, state, action, next_state):
+        deceleration = 1.0
+        if self._deceleration_zones:
+            deceleration = self._deceleration()
+
+        position = state + (deceleration * action)
+        dist, _ = self._sample_noise(position)
+        log_prob = dist.log_prob(tf.stop_gradient(next_state))
+        return log_prob
 
     def _reward_fn(self, state, action, next_state):
+        # pylint: disable=invalid-unary-operand-type
         goal = tf.constant(self._end, name="goal")
-        return -tf.norm(state - goal)  # pylint: disable=invalid-unary-operand-type
+        return -tf.norm(state - goal, axis=-1)
```

Comment:
This assumes that the last dimension of `state` represents the size of the state representation... So, if you pass a state with shape (batch_size, m) the result will be of shape (batch_size,). If you had more "batch" dimensions, it would work fine as well.